### ISSUE TITLE
`<Cmd>` keys not handled properly when used with `--remote-send`

### ISSUE THREAD

**Issue by Bekaboo** [neovim/neovim#31238]
### Problem

`<Cmd>` keys are not handled properly when used with `--remote-send`. Keys wrapped in between `<Cmd>` and `<CR>` can be interpreted as if they are typed in normal mode instead of part of a command.

For example, if an nvim instance is in the middle of a `gi` keymap, after pressing `g`, if we send it `<Cmd>if v:true | echo 'msg from remote' | endif<CR>`, it will insert `f v:true | echo 'msg from remote` in the current buffer.

### Steps to reproduce

1. `nvim --clean` to enter nvim
2. `:echo v:servername` to find the listen address
3. Press `g` to put nvim in the middle of the `gi` keymap
4. From another terminal, run `nvim --clean --headless --server <servername> --remote-send "<Cmd>if v:true | echo 'msg from remote' | endif<CR>" +'qa!'`
5. String "f v:true | echo 'msg from remote' | endif" is inserted in the buffer

### Expected behavior

nvim should execute the command instead of entering insert mode on the first 'i' and insert the rest of the command in the buffer.

### Nvim version (nvim -v)

v0.10.2

### Vim (not Nvim) behaves the same?

N/A

### Operating system/version

Linux 6.11.6

### Terminal name/version

alacritty 0.14.0 + tmux next-3.6

### $TERM environment variable

tmux-256color

### Installation

AUR

**Comment by justinmk** [comment]
`--remote` needs a complete overhaul. https://github.com/neovim/neovim/pull/18414

**Comment by zeertzjq** [comment]
I don't think this is related to `--remote-send`. If any RPC client sends this via `nvim_input` this will happen.

**Comment by zeertzjq** [comment]
Related #7350
